src/types.ts

```diff
@@ -28,9 +28,11 @@ export type ConversationHistory = {
   lastSaved: number;
   tokensRemaining: number;
 };
+export type ConversationsToGroup =
+  | ConversationHistories
+  | SearchableChatHistories;
 
-export type GroupedConversations = Record<string, ConversationHistory[]>;
-
+export type GroupedConversations = Record<string, ConversationsToGroup[]>;
```

Comment:
Extend this type to include `SearchableChatHistories` because all we are really concerned with is the shared `lastSaved` property in the grouping/sorting function.

The different components that these arrays will map to (ChatHistoryItems and SearchModal "rows") will handle rendering other elements, like `title` and message content